### DISCUSSION TITLE
onFocust not working on controller

### DISCUSSION THREAD

**Discussion by vinodhum** [react-hook-form/react-hook-form#2195] [Q&A]
Hi,
onFocus is not triggering on controller 
`  <Controller
        as={Input}
        control={control}
        name={name}
        type={type}
        label={label}
        onBlur={onBlur}
        onFocus={onFocus}
        defaultValue={defaultValue}
        error={!!errors[name]}
        errorText={errors[name]?.message}
        width="100%"
        rules={validation}
        disabled={disabled}
      `/>``

if I use custom Input directly it is working
`
<Input name="somename" type="text" onFocus={onFocus} />
`
what i am doing wrong with using controller ? on blure and on click is working with controller but onFocus is not tirggering.

Thank You

**Comment by vinodhum** [comment]
running into couple of more issues.

How to set custom validation ?
I have array of objects
list = [
{
nickname : 'mynickname'
}
]

I want to validate on user input that, nickname input should not take duplicate nickames
where validation is {required : true}
I am using rules={validation}
on top of validation how to add custome validation ?

I tried using onBlur and handling duplicate logic and setting error, but for some reason setError is not working

**Reply by bluebill1049** [reply]
pretty sure, it's working, i made a video last weekend with it using React native: https://www.youtube.com/watch?v=FNnSqP2rWis&feature=emb_logo
